### PR TITLE
fix(): use BroadcastMessage to reload investigation table

### PR DESCRIPTION
**Pull request by EmilyLip**

### CODE REVIEW

File: client/src/components/App/Content/InvestigationForm/useInvestigationForm.ts [hunk 1]
```diff
@@ -223,7 +226,11 @@ const useInvestigationForm = (): useInvestigationFormOutcome => {
             showConfirmButton: false
         });
         timeout(LandingPageTimer).then(() => {
-            setIsInInvestigation(false);
+            const broadcastMessage : BroadcastMessage = {
```
Comment (NaamanTsur) [inline client/src/components/App/Content/InvestigationForm/useInvestigationForm.ts:229]: You should define the `BroadcastChannel` here instead.

File: client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts [hunk 2]
```diff
@@ -110,6 +111,11 @@ const useInvestigationTable = (parameters: useInvestigationTableParameters): use
     const axiosInterceptorId = useSelector<StoreStateType, number>(state => state.investigation.axiosInterceptorId);
     const isInInvestigations = useSelector<StoreStateType, boolean>(state => state.isInInvestigation);
 
+    const bc = new BroadcastChannel(BC_TABS_NAME);
```
Comment (NaamanTsur) [inline client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts:114]: You should define the `BroadcastChannel` in a useEffect so it is initialized only once

**Review by NaamanTsur — DISMISSED** [review]

File: client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/useInvestigatedPersonInfo.ts [hunk 3]
```diff
@@ -27,14 +27,19 @@ const useInvestigatedPersonInfo = (): InvestigatedPersonInfoOutcome => {
...
-
+    
     const handleInvestigationFinish = async () => {
         alertSuccess('בחרת לצאת מהחקירה לפני השלמתה! הנך מועבר לעמוד הנחיתה', {
             timer: 1750,
             showConfirmButton: false
         });
         timeout(1500).then(() => {
-            setIsInInvestigation(false);
+            const bc = new BroadcastChannel(BC_TABS_NAME);
```
Comment (eladyashar) [inline client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/useInvestigatedPersonInfo.ts:37]: use meaning full name - cloasingInvastigationBroadcatChannel for exemple

File: client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/useInvestigatedPersonInfo.ts [hunk 4]
```diff
@@ -27,14 +27,19 @@ const useInvestigatedPersonInfo = (): InvestigatedPersonInfoOutcome => {
...
+            const broadcastMessage : BroadcastMessage = {
```
Comment (eladyashar) [inline client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/useInvestigatedPersonInfo.ts:38]: use meaningful name -  cloasingInvastigationBroadcastMassage for example

Comment (eladyashar) [reply client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/useInvestigatedPersonInfo.ts:38]: relevant for all usage in bc variable

File: client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts [hunk 5]
```diff
@@ -158,6 +161,9 @@ const useInvestigationTable = (parameters: useInvestigationTableParameters): use
         fetchAllInvestigationStatuses();
         fetchAllDesksByCountyId();
         startWaiting();
+        bc.current.onmessage = function (ev) {
```
Comment (RoeiRom) [inline client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts:164]: change to arrow function

**Review by RoeiRom — CHANGES_REQUESTED** [review]

File: client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/useInvestigatedPersonInfo.ts (hunk 3 above)
Comment (EmilyLip) [reply client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/useInvestigatedPersonInfo.ts]: DONE

File: client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/useInvestigatedPersonInfo.ts (hunk 4 above)
Comment (EmilyLip) [reply client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/useInvestigatedPersonInfo.ts:38]: DONE

File: client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts [hunk 6]
```diff
@@ -158,6 +161,9 @@ const useInvestigationTable = (parameters: useInvestigationTableParameters): use
         fetchAllInvestigationStatuses();
         fetchAllDesksByCountyId();
         startWaiting();
+        WindowTabsBroadcatChannel.current.onmessage = function (ev) {
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts:164]: ev stands for event?

**Review by RoeiRom — APPROVED** [review]

File: client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts (hunk 6 above)
Comment (EmilyLip) [reply client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts]: yes

File: client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts (hunk 5 above)
Comment (EmilyLip) [reply client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts]: DONE

**Review by RoeiRom — APPROVED** [review]

**Review by eladyashar — DISMISSED** [review]

File: client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts (hunk 6 above)
Comment (ItayBenMoshe) [reply client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts]: so maybe change the name to event

Comment (EmilyLip) [reply client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts]: I think it's clear enough, but changed it anyway

**Review by ItayBenMoshe — APPROVED** [review]

**Review by RoeiRom — APPROVED** [review]

**Pull request merged**